core/src/main/java/org/apache/iceberg/rest/requests/CommitTransactionRequestParser.java

```diff
@@ -0,0 +1,126 @@
+import java.io.IOException;
+import java.util.List;
+import org.apache.iceberg.MetadataUpdate;
+import org.apache.iceberg.MetadataUpdateParser;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.catalog.TableIdentifierParser;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.util.JsonUtil;
+
+public class CommitTransactionRequestParser {
+  private static final String TABLE_CHANGES = "table-changes";
+  private static final String IDENTIFIER = "identifier";
+  private static final String REQUIREMENTS = "requirements";
+  private static final String UPDATES = "updates";
+
+  private CommitTransactionRequestParser() {}
+
+  public static String toJson(CommitTransactionRequest request) {
+    return toJson(request, false);
+  }
+
+  public static String toJson(CommitTransactionRequest request, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(request, gen), pretty);
+  }
+
+  public static void toJson(CommitTransactionRequest request, JsonGenerator gen)
+      throws IOException {
+    Preconditions.checkArgument(null != request, "Invalid commit tx request: null");
+
+    gen.writeStartObject();
+    gen.writeFieldName(TABLE_CHANGES);
+    gen.writeStartArray();
+
+    for (UpdateTableRequest tableChange : request.tableChanges()) {
+      gen.writeStartObject();
+
+      gen.writeFieldName(IDENTIFIER);
+      TableIdentifierParser.toJson(tableChange.identifier(), gen);
+
+      gen.writeArrayFieldStart(REQUIREMENTS);
+      for (UpdateTableRequest.UpdateRequirement updateRequirement : tableChange.requirements()) {
+        UpdateRequirementParser.toJson(updateRequirement, gen);
+      }
+      gen.writeEndArray();
+
+      gen.writeArrayFieldStart(UPDATES);
+      for (MetadataUpdate metadataUpdate : tableChange.updates()) {
+        MetadataUpdateParser.toJson(metadataUpdate, gen);
+      }
+      gen.writeEndArray();
+
+      gen.writeEndObject();
+    }
+
+    gen.writeEndArray();
+    gen.writeEndObject();
+  }
+
+  public static CommitTransactionRequest fromJson(String json) {
+    return JsonUtil.parse(json, CommitTransactionRequestParser::fromJson);
+  }
+
+  public static CommitTransactionRequest fromJson(JsonNode json) {
+    Preconditions.checkArgument(null != json, "Cannot parse commit tx request from null object");
+
+    List<UpdateTableRequest> tableChanges = Lists.newArrayList();
+    JsonNode changes = JsonUtil.get(TABLE_CHANGES, json);
+
+    Preconditions.checkArgument(
+        changes.isArray(), "Cannot parse commit tx request from non-array: %s", changes);
+
+    for (JsonNode node : changes) {
+      TableIdentifier identifier = null;
+      List<UpdateTableRequest.UpdateRequirement> requirements = Lists.newArrayList();
+      List<MetadataUpdate> updates = Lists.newArrayList();
+
+      if (node.hasNonNull(IDENTIFIER)) {
```

Comment:
Is the plan to refactor this into an `UpdateTableRequestParser`? Seems strange to write this but not as a separate class for that request.

Reply:
yes that will be refactored into `UpdateTableRequestParser` in https://github.com/apache/iceberg/pull/7750